src/backend/api/meals.js

```diff
@@ -1,15 +1,123 @@
 const express = require("express");
+const { limit, sum } = require("../database");
 const router = express.Router();
 const knex = require("../database");
 
 router.get("/", async (request, response) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    let meals = await knex("meal");
+
+    //Returns all meals that are cheaper than maxPrice.
+    if(request.query.maxPrice !== undefined) {
+      const reqPrice = Number(request.query.maxPrice);
+      if(isNaN(reqPrice)) {
+        response.status(400).send({"message": "MaxPrice should be a number"});
```

Comment:
You should add a `return` here or on the line below otherwise if you have an invalid `maxPrice` and invalid `availableReservations` for instance, then you will try to send 2 responses and that would fail.